src/diffusers/models/adapter.py

```diff
@@ -0,0 +1,307 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+import torch.nn as nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from .modeling_utils import ModelMixin
+from .resnet import Downsample2D
+
+
+class ResnetBlock(nn.Module):
```

Comment:
Would it be possible to use our existing `ResnetBlock2D`? It looks very similar

Reply:
I did think of that before, but the existence of `ResnetBlock2D.norm1`, `ResnetBlock2D.norm2`, and the time step embedding argument in the forward method prevented me from using it to implement the adapter,

Reply:
the time embedding is optional, feel free to make a small change to the block definition where time_embedding_norm can take the value "no_norm" which when set will set norm1 and norm2 to None and then they're skipped in the forward

Reply:
@HimariO any progress on replacing the resnet block? happy to help if needed

Reply:
I later spot some more differences:
1. Adapter sometimes uses kernel size 1 in the second conv2d module of the ResNet block, so we will need to add one(or two, since there are two conv2d in the ResNet block) more parameters to the `ResnetBlock2D`
2. Adapter use the "conv2d -> activation -> conv2d" pattern, and the `ResnetBlock2D` use "activation -> conv2d -> activation -> conv2d" pattern. Not sure there is any good way to bypass `ResnetBlock2D`'s first activation function.

Not sure it is a good idea to make those changes to `ResnetBlock2D` or not, some help/suggestions here will be great :)

Reply:
I'll try to look into it tomorrow! I think it'd be totally fine though to slighly adapt the existing ResnetBlock